pandas/tests/test_window.py

```diff
@@ -423,6 +423,27 @@ def test_constructor_with_timedelta_window(self):
             expected = df.rolling('3D').sum()
             tm.assert_frame_equal(result, expected)
 
+    def test_constructor_with_timedelta_window_and_minperiods(self):
+        # GH 15305
+        n = 10
+        df = pd.DataFrame({'value': np.arange(n)},
+                          index=pd.date_range('2015-12-24',
+                                              periods=n,
+                                              freq="D"))
+        expected_data = np.append([np.NaN, 1.], np.arange(3., 27., 3))
+        for window in [timedelta(days=3), pd.Timedelta(days=3)]:
+            result_roll_sum = df.rolling(window=window, min_periods=2).sum()
+            result_roll_generic = df.rolling(window=window,
+                                             min_periods=2).apply(sum)
+            expected = pd.DataFrame({'value': expected_data},
+                                    index=pd.date_range('2015-12-24',
+                                                        periods=n,
+                                                        freq="D"))
+            tm.assert_frame_equal(result_roll_sum, expected)
+            tm.assert_frame_equal(result_roll_generic, expected)
+            expected = df.rolling('3D', min_periods=2).sum()
```

Comment:
This tests when window == '3D'. I'd better put it together with [timedelta(days=3), pd.Timedelta(days=3)]